### ISSUE TITLE
Issues related to TypeScript types.

### ISSUE THREAD

**Issue by Jamerrone** [datocms/react-datocms#54]
Hey there 👋,

Thanks a lot for the wonderful lib.

But I am running into some issues trying to get it working with TypeScript.
While the code below works fine, I do get the following TS error message: `Object is of type 'unknown'.` at: `data={record.src.responsiveImage}`.

``` typescript
const StructuredText = ({ data }: { data: StructuredTextGraphQlResponse }) => {
  return (
    <DatoStructuredText
      renderBlock={({ record }) => {
        switch (record._modelApiKey) {
          case 'ti_image':
            return <DatoImage data={record.src.responsiveImage} />
          default:
            return null
        }
      }}
      data={data}
    />
  )
}
```

Which makes totally sense:
```typescript
export declare type Record = {
    __typename: string;
    id: string;
} & {
    [prop: string]: unknown;
};
```

However, is there any workaround for this?

I also have a second issue. While I can move `renderBlock` to its own function, I can't do the same for `renderLinkToRecord`. This is because the `children` type is not currently exported. The record type, however, is exported under an alias: `StructuredTextGraphQlResponseRecord`. Is this something that can be added as well?

Thanks a lot for your time, and again, I really appreciate the lib.

**Comment by stefanoverna** [comment]
Thank you!

Here you can see how you can define exactly which blocks/links are you expecting for a particular `StructuredTextGraphQlResponse`:

https://github.com/datocms/react-datocms/blob/master/src/StructuredText/__tests__/index.test.tsx#L146

If you do so, the type for the `record` argument on `renderBlock` will narrow down, and you'll be able to get rid of the TS error message.

---

Just released v3.0.11 which exports the `RenderInlineRecordContext`, `RenderRecordLinkContext` and `RenderBlockContext` types!

**Comment by Jamerrone** [comment]
Thank you so much! Awesome.
